lib/cchain/provider/abci.go

```diff
@@ -280,3 +315,194 @@ func (a rpcAdaptor) Invoke(ctx context.Context, method string, req, resp any, _
 func spanName(endpoint string) string {
 	return "cprovider/" + endpoint
 }
+
+// Finds the height at which the requested offset can be found in state.
+func searchOffsetInHistory(ctx context.Context, client ABCIClient, cl atypes.QueryClient, chainVer xchain.ChainVersion, fromOffset uint64) (int, error) {
```

Comment:
godoc comments have a strict format, you must start with the name of the function/type:

```
// searchOffsetInHistory searches the consensus state history and 
// returns the historical consensus block height that contains an approved attestation
// for the provided chain version and fromOffset.
```

Reply:
we always use `uint64` for heights and offsets, lets stick to that #consistencyFTW